CMakeLists.txt

```diff
@@ -44,6 +44,11 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
   endif()
 endif()
 
+if (CMAKE_HOST_SYSTEM_VERSION MATCHES ".*-Microsoft")
```

Comment:
This won't match for non-WSL builds on Windows?

Reply:
According to the CMAKE docs, it's only available on platforms with uname: https://cmake.org/cmake/help/v3.4/variable/CMAKE_HOST_SYSTEM_VERSION.html?highlight=uname

> A numeric version string for the system. On systems that support uname, this variable is set to the output of uname -r. On other systems this is set to major-minor version numbers.

On regular Windows that would 6.1